src/messages/textDocument_document.cc

```diff
@@ -165,88 +152,70 @@ void MessageHandler::textDocument_documentSymbol(JsonReader &reader,
     std::sort(result.begin(), result.end());
     reply(result);
   } else if (g_config->client.hierarchicalDocumentSymbolSupport) {
-    std::unordered_map<SymbolIdx, std::unique_ptr<DocumentSymbol>> sym2ds;
-    std::vector<std::pair<std::vector<const void *>, DocumentSymbol *>> funcs,
-        types;
+    std::vector<ExtentRef> syms;
+    syms.reserve(file->symbol2refcnt.size());
+
+    /**
+     * with 2 ranges that start at the same Position, we want the wider one
+     * first (swap lhs/rhs)
+     *
+     * we use range for start to start at symbol name
+     * issue with: int i, j, k; otherwise
+     */
+    auto sym_cmp = [](ExtentRef const &lhs, ExtentRef const &rhs) {
+      return lhs.range.start < rhs.range.start ||
```

Comment:
I am currently using
```cpp
                return std::tie(lhs.range.start, rhs.extent.end) <
                       std::tie(rhs.range.start, lhs.extent.end);
```

but why not
```cpp
                return std::tie(lhs.extent.start, rhs.extent.end) <
                       std::tie(rhs.extent.start, lhs.extent.end);
```

Reply:
This is the same problem as under: https://github.com/MaskRay/ccls/pull/674/commits/48b51507fdfeda0d4a23014b08e2683d8ddc72d0#r628918008
`extent.start` points to the type, and `range.start` points to the variable name

to put i then j then k flattened we need different start position

Reply:
Sorry, it's there to sort `int i, j, k` in the correct order, or else it keeps the order of `file->symbol2refcnt` as `extent.start` of i, j and k is the same value (points to `int`) but `range.start` points to the variable name.

Reply:
OK, you meant global variables `int i, j, k;` I understand that now. You are right, sorting by `range.start` can fix the nested k,j,i problem.

Fixed. Thanks!

Reply:
or static/member variables